src/main/java/Jay.java

```diff
@@ -0,0 +1,153 @@
+import command.Command;
+import command.InvalidCommandException;
+import parser.Parser;
+import storage.DataIOException;
+import task.InvalidTaskException;
+import task.Task;
+
+import java.util.Scanner;
+
+/**
+ * Jay is a personal assistant chatbot that helps users to keep track of various tasks.
+ */
+public class Jay {
+    public static void main(String[] args) {
+        Jay jay = new Jay("Jay");
+        jay.start();
+    }
+
+    private final TaskList tasks;
+    private final Ui ui;
+
+    public Jay(String name) {
+        this.tasks = new TaskList("tasks.txt");
+        this.ui = new Ui(name);
+    }
+
+    /**
+     * Starts the chatbot.
+     */
+    public void start() {
+        ui.greet();
+        Scanner scanner = new Scanner(System.in);
+
+        while (scanner.hasNext()) {
+            String command = scanner.nextLine();
+            if (command.equals("bye")) {
+                ui.quit();
+                break;
+            } else {
+               ui.output(this.processCommand(command));
+            }
+        }
+    }
+
+    /**
+     * Processes the command and returns the response.
+     *
+     * @param commandStr The command to be processed.
+     * @return The response to the command. Show to the user by the chatbot.
+     */
+    private String processCommand(String commandStr) {
+        Command command = new Command(commandStr);
+        Command.CommandType commandType = command.getCommandType();
+
+        try {
+            switch (commandType) {
```

Comment:
Should your switch-case be in the following form?
```Java
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}
```
You can go to Settings -> Editor -> Code Style -> Java
Choose *Wrapping and Braces* tab and uncheck *Indent 'case' branches* under **'switch' statement/expression**